src/systems/ec/implementations/react/packages/button/Button.jsx

```diff
@@ -24,6 +26,8 @@ const Button = ({
     >
       <span className="ecl-button__container">
         <span className="ecl-button__label">{label}</span>
+        {icon &&
+          icon.icon && <Icon {...icon} extraClasses="ecl-button__icon" />}
```

Comment:
is it necessary to check `icon.icon`?

Reply:
Yes, it is, as icon has some default value (for size), so object is net empty